File: internhub-frontend/src/models/Internship.js
import { Company, StudyArea } from "./index";

export class Internship {
  constructor({
    id = "",
    studyAreaId = "",
    studyArea = null,
    companyId = "",
    company = null,
    name = "",
    description = "",
    address = "",
    startDate = "",
    endDate = "",
    applicationsCount = 0,
    isApplied = false,
  }) {
    this.id = id;
    this.studyAreaId = studyAreaId;
    this.studyArea = studyArea;
    this.companyId = companyId;
    this.company = company;
    this.name = name;
    this.description = description;
    this.address = address;
    this.startDate = startDate;
    this.endDate = endDate;
    this.applicationsCount = applicationsCount;
    this.isApplied = isApplied;
  }

  static fromJson(json) {
    return new Internship({
      id: json["Id"],
      studyAreaId: json["StudyAreaId"],
      companyId: json["CompanyId"],
      name: json["Name"],
      description: json["Description"],
      address: json["Address"],
      startDate: json["StartDate"],
      endDate: json["StartDate"],
      company: json["Company"] ? Company.fromJson(json["Company"]) : null,
      studyArea: json["StudyArea"]
        ? StudyArea.fromJson(json["StudyArea"])
        : null,
      applicationsCount: json["ApplicationsCount"],
      isApplied: json["IsApplied"] ? json["IsApplied"] : false,
    });
  }
}

File: InternHub/InternHub.Model/Internship.cs
using InternHub.Model.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InternHub.Model
{
    public class Internship : BaseModel, IInternship
    {
        public Guid StudyAreaId { get; set; }
        public StudyArea StudyArea { get; set; }
        public string CompanyId { get; set; }
        public Company Company { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Address { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public long ApplicationsCount { get; set; }
    }
}
